superset-frontend/src/dashboard/components/FiltersBadge/selectors.ts

```diff
@@ -147,11 +144,10 @@ export const selectIndicatorsForChart = (
   chartId: number,
   filters: { [key: number]: Filter },
   datasources: { [key: string]: Datasource },
-  charts: any,
+  chart: any,
 ): Indicator[] => {
-  const chart = charts[chartId];
-  // no indicators if chart is loading
-  if (chart.chartStatus === 'loading') return [];
+  // no indicators before chart is rendered
+  if (chart.chartStatus !== 'rendered') return [];
```

Comment:
This change (and the other one for native filters) removes the indicator if no results are returned:

### BEFORE
![image](https://user-images.githubusercontent.com/33317356/123759998-1c509000-d8c9-11eb-87c9-865b8e327def.png)

### AFTER
![image](https://user-images.githubusercontent.com/33317356/123760129-3ab68b80-d8c9-11eb-8a3d-642fb6df871c.png)

This should probably either be left unchanged, or changed to 
```js
  if (!['rendered', 'success'].includes(chart.chartStatus) return [];
```

Reply:
Thanks for finding this! I used a different solution so that we don't rerun the function when chartStatus changes from `success` to `rendered`